### ISSUE TITLE
Storage: Explain what "Store Passphrase" does

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#11824]
Page: Storage

The Format dialog has a "Store Passphrase" option:

![screenshot](https://user-images.githubusercontent.com/3228183/57455102-11fb5400-7273-11e9-954c-db867b6c12c0.png)

It's not very clear what that does -- store it in the file system, store it in Cockpit, is it being stored in cleartext or encrypted, what is that good for, etc. This should at least get an explanatory tooltip?

**Comment by andreasn** [comment]
Oh yeah, good call. I was wondering about that one as well.
Is there any documentation on this?

**Comment by xpseudonym** [comment]
Err, at least if someone could answer the question here it would allow people to search - and find - the answer...
What does it do?

**Comment by xpseudonym** [comment]
OT: explain what the mounting defaults are...

**Comment by marusak** [comment]
I guess that it should be documented [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/managing_systems_using_the_rhel_8_web_console/managing-storage-devices-in-the-web-console_system-management-using-the-rhel-8-web-console#formatting-logical-volumes-in-the-web-console_using-the-web-console-for-configuring-lvm-logical-volumes) and then we can point to it, the same way we did in with network bonds in https://github.com/cockpit-project/cockpit/pull/13507

And I guess only @mvollmer actually knows what it does :)

**Comment by xpseudonym** [comment]
:o)

**Comment by andreasn** [comment]
@marusak I'm not sure documentation is enough. If we can find a clearer label, that would help a lot.

**Comment by martinpitt** [comment]
@mvollmer: As it came up recently again: Can you please describe (for posterity) what this is about? To me it sounds *really* scary.. Should we even offer that option in the UI?

**Comment by mvollmer** [comment]
This is a feature that Cockpit shared with GNOME Disks. It is a interface to the third field of the entries in /etc/crypttab:

    The third field specifies the encryption password. If the field is not present or the password is set to
    "none" or "-", the password has to be manually entered during system boot. Otherwise, the field is 
    interpreted as an absolute path to a file containing the encryption password.  [...]

Storing passphrases in this way makes sense if your /etc directory is secure, such as when / itself is encrypted.


GNOME Disks doesn't offer to store the passphrase at all when creating a encrypted volume.  Later, when editing the encryption options of a encrypted volume, there is a field for a passphrase:

![image](https://user-images.githubusercontent.com/3228183/78338994-6d75b500-759c-11ea-9881-20fa5bf64520.png)

This corresponds to the "Stored passphrase" field in the Encryption tab in Cockpit:

![image](https://user-images.githubusercontent.com/3228183/78339185-bf1e3f80-759c-11ea-9302-c16d55358956.png)

**Comment by mvollmer** [comment]
So what about removing the "Store passphrase" option from the Format dialog?

We can add a lot of text into the "Stored passphrase" dialog to explain what it does.

![image](https://user-images.githubusercontent.com/3228183/78340301-867f6580-759e-11ea-8112-60dacef1777e.png)

**Comment by martinpitt** [comment]
The good side is that it exposes the feature less, it doesn't seem to be one that we should recommend. If you need automatic unlocking in the server world, then clevis/tang seems to be a better approach IMHO.

The bad side is that you can't define the wanted behaviour on creation, but  maybe in light of the former that's actually a feature :)

So overall, :+1: for doing that. We could also not offer storing the passphrase at all for the root file system?

**Comment by mvollmer** [comment]
> If you need automatic unlocking in the server world, then clevis/tang seems to be a better approach IMHO.

Yes, especially the TPM2 pin is interesting for that. Cockpit doesn't support that yet.  Adding support would also more or less force us to add support for complex/nested Clevis policies.

**Comment by mvollmer** [comment]
> We could also not offer storing the passphrase at all for the root file system?

Yes.  Storing one would be harmless, it just wouldn't work.

**Comment by garrett** [comment]
Catching up with this issue — it seems it doesn't actually need design? It seems like it should go away, in favor of tang+clevis, right?
